immuno_graphical/verification/immuno_graphical_verification_code.R

```diff
@@ -0,0 +1,470 @@
+box_plot("bindSpike","Placebo")
+box_plot("bindRBD","Placebo")
+box_plot("pseudoneutid50","Placebo")
+box_plot("pseudoneutid80","Placebo")
+
+
+# baseline Negative placebo arm
+ggplot(dat.long.twophase.sample%>%
+         filter(Trt == "Placebo", 
+                Bserostatus == "Baseline Neg") , aes(y = Day57)) +
+  stat_boxplot(aes(x=assay),
+               geom = "errorbar", 
+               width = 0.5) +  
+  geom_boxplot(aes(x=assay))  +
+  ylim(0,10)
+
+# baseline Negative vaccine arm
+
+ggplot(dat.long.twophase.sample%>%
+         filter(Trt == "Vaccine", 
+                Bserostatus == "Baseline Neg"), aes(y = Day57)) +
+  stat_boxplot(aes(x=assay),
+               geom = "errorbar", 
+               width = 0.5) +  
+  geom_boxplot(aes(x=assay))  +
+  ylim(0,10)
+
+
+
+#Spaghetti plots of Ab markers over time: baseline negative vaccine + placebo arm
+Spaghetti_plot <- function(assay1,Trt1){
+  Spaghettiplot_data <- dat.long.twophase.sample %>%
+    filter(assay == assay1,
+           Bserostatus == "Baseline Neg",
+           Trt == Trt1) 
+  write.csv(data.frame(summary=names(summary(Spaghettiplot_data$Day57)),value=matrix(summary(Spaghettiplot_data$Day57))),
+            here(paste("immuno_graphical/verification/output/output_tester/spaghetti_plots_summary_day57_trt_",Trt1,"_",assay1,".csv",sep="")),
+            row.names = FALSE     
+  )
+  write.csv(data.frame(summary=names(summary(Spaghettiplot_data$Day29)),value=matrix(summary(Spaghettiplot_data$Day29))),
+            here(paste("immuno_graphical/verification/output/output_tester/spaghetti_plots_summary_day29_trt_",Trt1,"_",assay1,".csv",sep="")),
+            row.names = FALSE     
+  )
+  write.csv(data.frame(summary=names(summary(Spaghettiplot_data$Day57)),value=matrix(summary(Spaghettiplot_data$Day57))),
+            here(paste("immuno_graphical/verification/output/output_tester/spaghetti_plots_summary_day57_trt_",Trt1,"_",assay1,".csv",sep="")),
+            row.names = FALSE     
+  )
+  write.csv(data.frame(summary=names(summary(Spaghettiplot_data$B)),value=matrix(summary(Spaghettiplot_data$B))),
+            here(paste("immuno_graphical/verification/output/output_tester/spaghetti_plots_summary_B_trt_",Trt1,"_",assay1,".csv",sep="")),
+            row.names = FALSE     
+  )
+}
+Spaghetti_plot("bindSpike","Vaccine")
+Spaghetti_plot("bindRBD","Vaccine")
+Spaghetti_plot("pseudoneutid50","Vaccine")
+Spaghetti_plot("pseudoneutid80","Vaccine")
+Spaghetti_plot("bindSpike","Placebo")
+Spaghetti_plot("bindRBD","Placebo")
+Spaghetti_plot("pseudoneutid50","Placebo")
+Spaghetti_plot("pseudoneutid80","Placebo")
+
+
+
+Spaghetti_plot_neg <- read.csv(
+  here("immuno_graphical/verification/input/output_primary_programmer/spaghetti_plot_data_BaselineNeg.csv")
+) %>% select(Ptid) %>% unique()
+
+Spaghetti_plot_pos <- read.csv(
+  here("immuno_graphical/verification/input/output_primary_programmer/spaghetti_plot_data_BaselinePos.csv")
+) %>% select(Ptid) %>% unique() 
```

Comment:
Could you provide context as to why you need to read this in?

Reply:
That's because the subjects plotted in the spaghetti plot were randomly selected by original programmer, so I need to get the list of Ptids that original programmer used to make sure the data was plotted correctly.

Reply:
👍 just add that into a comment so its recorded for posterity (ie when another programmer like me is looking and doesn't have context or you to ask)